escape_backrooms.asl

```diff
@@ -15,65 +20,78 @@ Documentation Notes
 */
 
 state("Backrooms-Win64-Shipping", "3.0+")
 {
-	bool isLoading		: 0x04C5F398, 0xD28, 0x320; // sets to true when starting to fade to black, false when loading screen finishes
-	bool justLoaded		: 0x4C3CFED; // briefly true when loading screen finishes
+	bool isLoading1		: 0x04C5F398, 0xD28, 0x320; // sets to true when starting to fade to black, false when loading screen finishes
+	bool isLoading2		: 0x04C5F998, 0x8, 0x60, 0x320; // sets to true when starting to fade to black, false when loading screen finishes, backup
+	bool wasLoading1	: 0x4C3CFED; // 0 when done loading, 0-2 (maybe higher?) when loading
+	bool wasLoading2	: 0x4C3CFF0; // 0 when done loading, > 0 when loading, backup
 	bool multiplayer	: 0x4760BE8; // set to true when in multiplayer is detected (including simply being in a multiplayer lobby)
 }
 
 state ("Backrooms-Win64-Shipping", "2.9")
 {
 	bool multiplayer	: 0x45FA838; // detects co-op lobby. other candidates: 0x45FA834 != 0 | 0x45FA83C = 4; != 0 | 0x45FA840 = 1 != 2
-	long level		: 0x49B0968; // 13194139536091 is always start, 13194139536054 is main menu, seems inconsistent for most other levels though
+	long level			: 0x49B0968; // 13194139536091 is always start, 13194139536054 is main menu, seems inconsistent for most other levels though
 	bool isLoading		: 0x49AD8C4; // 33 C0 0F 57 C0 F2 0F 11 05 and 89 43 60 8B 05
 }
 
 state ("Backrooms-Win64-Shipping", "2.3") // offset -180 from 2.9
 {
-	bool multiplayer   	: 0x45FA6F8; // detects co-op lobby. other candidates: 0x45FA834 != 0 | 0x45FA83C = 4; != 0 | 0x45FA840 = 1 != 2
-	long level		: 0x49B07E8; // 13194139536090 is always start, 13194139536054 is main menu, seems inconsistent for most other levels though
+	bool multiplayer	: 0x45FA6F8; // detects co-op lobby. other candidates: 0x45FA834 != 0 | 0x45FA83C = 4; != 0 | 0x45FA840 = 1 != 2
+	long level			: 0x49B07E8; // 13194139536090 is always start, 13194139536054 is main menu, seems inconsistent for most other levels though
 	bool isLoading		: 0x49AD744; // 33 C0 0F 57 C0 F2 0F 11 05 and 89 43 60 8B 05
 }
 
-startup
-{
-	settings.Add("29_addr", false, "Game Version 2.9 (Restart game to apply)");
-	settings.Add("23_addr", false, "Game Version 2.3 (Restart game to apply)");
-}
+//startup
+//{
+//	settings.Add("29_addr", false, "Game Version 2.9 (Restart game to apply)");
+//	settings.Add("23_addr", false, "Game Version 2.3 (Restart game to apply)");
+//}
 
 init
 {
 	vars.inLobby = false;
-	
-	print("[EtB Autosplitter] Escape the Backrooms Autosplitter loaded");
-	print("[EtB Autosplitter] Detected game version: " + (version == "" ? "Unknown, please contact the autosplitter authors for help!" : version));
 
 	IntPtr loadStartPtr = game.MainModule.BaseAddress;
 	IntPtr loadEndPtr = game.MainModule.BaseAddress;
 
-	if (settings["23_addr"]) {
-		version = "2.3";
+	if (modules.First().ModuleMemorySize == 83517440) // ModuleMemorySize for 2.3 and 2.9
+	{
+		if (settings["23_addr"])

```

Comment:
how can this be accessed if the startup settings.Add lines are commented out?

Reply:
Whoops! I must have forgotten to restore those lines after testing if I could detect the difference between v2.3 and v2.9.

I'll re-add that shortly.

Reply:
Re-added the `startup` in the latest commit!